library/RoleManager.unit.test.js

```diff
@@ -0,0 +1,33 @@
+const EventEmitter = require('events')
+
+class MockPracticeManager extends EventEmitter {
+  tick (activeTime, sessionTotal, overallTotal) {
+    this.emit('stopPractice', 1, activeTime, sessionTotal, overallTotal)
+  }
+}
+
+class RoleManager extends EventEmitter {
+  constructor ({ practiceManager }) {
+    super()
+    this.practiceManager = practiceManager
+    this.practiceManager.on('stopPractice', this.checkRole.bind(this))
+  }
+
+  checkRole (userId, activeTimeSec, sessionTotal, overallTotal) {
+    this.emit('roleAssigned', userId, '40 Hour Pracker')
+  }
+}
+
+test('assigns roles', () => {
+  const practiceManager = new MockPracticeManager()
+  const rm = new RoleManager({ practiceManager })
+  const eventValidatorSpy = jest.fn(function (userId, roleName) {
+    expect(userId).toBe(1)
+    expect(roleName).toBe('40 Hour Pracker')
+  })
+  rm.on('roleAssigned', eventValidatorSpy)
+
+  practiceManager.tick(2, 3, 4)
```

Comment:
I assume this is just for prototype purposes and we'll actually need positive/negative tests with a mock PracticeManager that has users close to but not yet reaching 40 hours.

Reply:
Yeah, this was just for messing around, to see if the interface would even work.